Bugsnag/RemoteConfig/Handler/BSGRemoteConfigHandler.m

```diff
@@ -105,16 +108,26 @@ - (BOOL)isRemoteConfigEnabled {
 }
 
 - (void)updateRemoteConfig {
+    @synchronized (self) {
+        self.isLoadingRemoteConfig = YES;
+    }
```

Comment:
Given how `isLoadingRemoteConfig` guards `updateRemoteConfig` shouldn't the `if` check also be here? That way the check itself can be guarded by the same `synchronized` block and remove any possible races?

Reply:
I mean that the "indirectly" is a problem, if we're using locks to ensure safety we shouldn't be relying on the caller to establish that lock, at least not without a clear naming convention (such as `updateRemoteConfigUnderLock`). The locks are re-entrant so re-locking them should be relatively cheap.